### PR TITLE
Updating package.json file for 0.4.2 release

### PR DESCRIPTION
**Pull request by lirantal**
Preparing for 0.4.2 release, updating relevant files.

### CODE REVIEW

**Comment by lirantal** [comment]
@codydaig any other updates we need?

**Comment by codydaig** [comment]
@lirantal Don't forget to update the bower.json.
Also, I would run the changelog and merge that in with this PR so it's in the release tag. 

I would like to vote to make 0.4.x a branch after the release but before we start merging 0.5 stuff in. I think there's a lot of people that picked up 0.4 and if there's any bugs, it gives us the opportunity to fix those bugs and release patches.

The generator does need to be updated, however, it's on hold until @ilanbiala hashes out a refactor for the built in `fs` to use some other module. There's a bunch of work that we still need to do to the generator, but it's on hold over a discussion over using `this.fs`versus the built in `fs`.

**Comment by lirantal** [comment]
@codydaig what is there to update in bower.js?

**Comment by codydaig** [comment]
@lirantal We must have removed it, but I swore there was a version in the bower.json as well. Glad it's gone now! :)

and I put the exact command that needs to get ran in the CONTRIBUTING.md document.

**Comment by lirantal** [comment]
@codydaig I did run it as follows `conventional-changelog -r 1 > CHANGELOG.md` although git shows a great diff between the former and new file and I suspected that something was wrong. Maybe it's just spaces or something else.

**Comment by ilanbiala** [comment]
@lirantal it looks for all the commits between 2 tags, so you would have to tag 0.4.2 first and then rerun this and you should see it work.

**Comment by codydaig** [comment]
@lirantal I will run the changelog today and make a PR to your branch for you. If there's something that I documented wrong, I will also correct that.

**Comment by codydaig** [comment]
@lirantal @ilanbiala I made a PR to @lirantal's branch with the updated changelog.

**Comment by lirantal** [comment]
Hey guys,
Was a bit busy with work, and some hackathon that was going on.

I'll get back to this really quick!

**Comment by lirantal** [comment]
weird, no idea why it's having conflicts on this branch while I can merge it with my master branch.

**Comment by lirantal** [comment]
had to merge from master to resolve conflicts.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,128 @@
+<a name="0.4.2"></a>
+## 0.4.2 (2015-11-11)
+
+
+* bug(readme): Remove Broken Live Demo Link ([9bfb312](https://github.com/meanjs/mean/commit/9bfb312))
```
Comment (ilanbiala) [inline CHANGELOG.md]: How come these aren't being parsed out into a subheading?

**Comment by ilanbiala** [comment]
@lirantal that's fine, just make sure to rebase and squash. I'm curious about why the changelog didn't come out with a subheading for bugs though. Ideas?

File: CHANGELOG.md (hunk 1 above)
Comment (lirantal) [inline CHANGELOG.md]: donnu, @codydaig ?

**Comment by codydaig** [comment]
@ilanbiala It did, but I removed them since there was only one or two things that fell under bugs. It only works with the new style and since the new style wasn't implemented until the very end most commits don't fall under a heading.

**Comment by ilanbiala** [comment]
Oh okay.

**Comment by lirantal** [comment]
@ilanbiala for whatever reason when I'm rebasing without a merge commit it is conflicting with master branch and we're unable to merge through Github. So I'll re-base again with a merge commit and merge for the 0.4.2 release.

File: CHANGELOG.md (hunk 1 above)
Comment (ilanbiala) [inline CHANGELOG.md]: @lirantal clarified, it's fine for this.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -36,3 +77,9 @@ Before you create a new Issue:
 ## Documentation (meanjs.org)
 
 The code for the documentation and the website are located in the meanjs/mean repo on the gh-pages branch. The website is hosted by github pages and generated using jekyll. See the README in the gh-pages branch for more information on how to get your dev enviorment set-up.
+
+## Generating a new Release
+
+For those who will be generating releases:
+* Don't forget to update the changelog!
+  * `conventional-changelog -r 1`
```
Comment (ilanbiala) [inline CONTRIBUTING.md]: This should have the proper CLI command that prepends it

Comment (codydaig) [inline CONTRIBUTING.md]: This is probably where the merge conflict is.

**Comment by codydaig** [comment]
LGTM

**Pull request merged**